### PR TITLE
Update deploying_with_k8s.md with AMD ROCm GPU example

### PR DESCRIPTION
**Pull request by AlexHe99**
Add the example of using AMD ROCm GPU

### CODE REVIEW

**Comment by DarkLight1337** [comment]
Can we make each step a section and have separate sub-sections for NVIDIA vs AMD GPUs?

File: docs/source/serving/deploying_with_k8s.md [hunk 1]
```diff
@@ -119,6 +119,79 @@ spec:
...
+            cpu: "6"
+            memory: 6G
+            amd.com/gpu: "1"
+        volumeMounts:
+        - name: cache-volume
+          mountPath: /root/.cache/huggingface
+        - name: shm
+          mountPath: /dev/shm
+```
+The full example is at https://github.com/ROCm/k8s-device-plugin/tree/master/example/vllm-serve.
```
Comment (DarkLight1337) [inline docs/source/serving/deploying_with_k8s.md:193]:
```suggestion
The full example is at [https://github.com/ROCm/k8s-device-plugin/tree/master/example/vllm-serve].
```

Also we should make this an actual link.

**Comment by AlexHe99** [comment]
> To better organize things, can we make each step a section and have separate sub-sections for NVIDIA vs AMD GPUs?

**Comment by AlexHe99** [comment]
> To better organize things, can we make each step a section and have separate sub-sections for NVIDIA vs AMD GPUs?

I commit new patches to split the sections of deployment.yaml to two sub-sections. Other steps like PVC, SVC are same for both of them.

**Review by DarkLight1337 — APPROVED** [review]
This looks good to me, thanks for improving the documentation!

**Pull request merged**